lib/TPP/Dialect/Tpp/BufferizableOpInterfaceImpl.cpp

```diff
@@ -260,17 +262,23 @@ getAliasingOpResultsBinaryImpl(Operation *op, OpOperand &opOperand,
   // If the rhs can bufferize in place with the result return the rhs.
   if (opOperand.getOperandNumber() == 1 &&
       opOperand.get().getType() == op->getResult(0).getType() &&
-      !isConstantVal(opOperand.get()))
+      !isConstantVal(opOperand.get())) {
     return {{op->getOpResult(0), BufferRelation::Equivalent,
              /*isDefinite=*/true}};
+  }
   // If the lhs can bufferize in place with the result return the lhs. Note
   // that if both can bufferize with the result we select the rhs first.
   if (opOperand.getOperandNumber() == 0) {
-    if (op->getOpOperand(1).get().getType() == op->getResult(0).getType())
+    // The lhs does not alias with op result if we can bufferize on the rhs.
+    if (op->getOpOperand(1).get().getType() == op->getResult(0).getType() &&
```

Comment:
```suggestion
    if (op->getOpOperand(1)->getType() == op->getResult(0).getType() &&
```

Reply:
not a ptr type, we cannot use `->`.

Reply:
Sorry, mistook `shared_ptr` semantics. Ignore me.